prime-router/docs/proposals/0020-universal-metadata-search/0020-universal-metadata-search.md

```diff
@@ -549,4 +549,208 @@ The query on just walking the report lineage suffers similarly just at a differe
+
+```mermaid
+graph TD  
+A(Sender Report A\n Item 1:+Flu\n Item 2:-Flu,+Covid) -- Convert\n Sender Transform --> B(Report B\n Item 1)
+A -- Convert\n Sender Transform --> C(Report C\n Item 2.1:-Flu) 
+A -- Convert\n Sender Transform --> Q(Report C\n Item 2.2:+Covid) 
+B -- Route to\nReceiver X\nReceiver Z--> D(Report D\nItem 1)
+C -- Route to\nReceiver Z --> E(Report E\nItem 2.1)
+Q -- Route to\nReceiver Y --> R(Report R\nItem 2.2)
+D -- Translate\nReceiver X --> F(Report F\nItem 1)
+D -- Translate\nReceiver Z --> G(Report G\nItem 1)
+E -- Translate\nReceiver X --> H(Report H\nItem 2.1)
+E -- Translate\nReceiver Z --> P(Report P\n Item 2.1)
+F -- Batch\nReceiver X --> J(Report J\nItem 1\nItem 2.1)
+H -- Batch\nReceiver X --> J
+G -- Batch\nReceiver Z --> K(Report K\nItem 1\nItem 2.1)
+P -- Batch\nReceiver Z --> K
+R -- Batch\nReceiver Y --> L(Report L\nItem 2.2)
+J -- Send\nReceiver X --> M(Report M\nItem 1\nItem 2.1)
+K -- Send\nReceiver Z --> N(Report N\nItem 1\nItem 2.1)
+L -- Send\nReceiver Y --> O(Report O\nItem 2.2)
+```
+
+
+#### Re-combining items
+
+```mermaid
+graph TD  
+A(Sender Report A\n Item 1:+Flu,+Covid) -- Convert\nSender Transform --> B(Report B\nItem 1.1:+Flu)
+A -- Convert\nSender Transform --> C(Report C\nItem 1.2:+Covid)
+B -- Route\nReceiver X --> D(Report D\nItem 1.1)
+C -- Route\nReceiver X --> E(Report E\nItem 1.2)
+D -- Translate --> F(Report F\nItem 1.1)
+E -- Translate --> G(Report G\n Item 1.2)
+F -- Merge --> H(Report H\n Item 1)
+G -- Merge --> H
+H -- Batch\nReceiver X --> I(Report I\nItem 1)
+I -- Send\nReceiver X --> J(Report J\n Item 1)
+```
+
+The graph above represents the use case where the initial report contains a single item with multiple results which is then split into multiple items (same concept as the previous graph), with the difference being that Receiver X is ultimately going to get both results.  In this situation, there now needs to be a new step in the pipeline `MERGE` which is going to take the two FHIR bundles (Item 1.1 and Item 1.2) that were split from the original reported Item 1 and combine them back so that the receiver gets just a single item in the report.
+
+## Potential Solution #3: Re-implement tracing reports/items
+
+There are many possible ways that this could be done and would be a substantial lift for the team and would likely only make sense to do if we had clear requirements on what kind of searching
+needs, the SLA that each searching requirement needs to be held to (i.e. an API the website consumes vs. a query someone on the team runs ad-hoc) and most importantly what should be the atomic
+unit that the system tracks.
+
+Recording some possible solutions that have been discussed previously as starting point if the decision is made to re-implement.
+
+- Store this data in elasticsearch
```

Comment:
Since the team has floated a bunch of interesting ideas around if we opted to do a greenfield approach, I've started capturing them here without flushing them out since as I mention I think we need firmer requirements before doing far down an implementation.

Reply:
agree